website/content/docs/concepts/request-limiter.mdx

```diff
@@ -0,0 +1,71 @@
+---
+layout: docs
+page_title: 'Request Limiter'
+description: >-
+  Vault provides an adaptive concurrency limiter to protect the Vault server
+  from overload.
+---
+
+# Request Limiter
+
+@include 'alerts/enterprise-only.mdx'
+
+@include 'alerts/beta.mdx'
+
+This document contains conceptual information about the **Request Limiter** and
+its user-facing effects.
+
+## Preventing overload
+
+The Request Limiter aims to prevent overload by proactively detecting latency
+and adapting the number of allowed in-flight requests.
+
+This is done in two phases at the beginning of an HTTP request:
+
+1. Consult the current number of allowed in-flight requests. If the new request
+would exceed this limit, immediately reject it, indicating that the client
+should retry later.
+
+2. If the request is allowed, begin a measurement of its latency, allowing the
+Request Limiter to calculate a new limit.
+
+## Resource constraints
+
+The Request Limiter intentionally focuses on preventing overload derived from
+resource-constrained operations on the Vault server. There are two known
+resource-constrained request paths in Vault:
+
+1. Back pressure from the storage backend, resulting in bufferbloat in the WAL
+system (enterprise) for `Write`-based HTTP methods.
+
+2. Back pressure from CPU over-utilization via PKI issue requests (specifically
+for RSA keys), resulting in failed heartbeats.
+
+Storage constraints can be accounted for by limiting logical requests according
+to their `http.Method`. We only limit requests with write-based methods, since
+these will result in storage Puts and exhibit the aforementioned bufferbloat.
+
+CPU constraints are accounted for using the same underlying library and
+technique; however, they require special treatment. The maximum number of
+concurrent pki/issue requests found in testing (again, specifically for RSA
+keys) is far lower than the minimum tolerable write request rate.
```

Comment:
```suggestion
CPU constraints are accounted for using the same underlying library and
technique; however, they the nature of the resource is different. The maximum number of concurrent PKI issue requests is directly linked to how many hardware CPU cores are available for the expensive computations. This number tends to be lower and more absolute than storage system throughput and latency so we need a separate limiter tuned to respond to the types of behaviour saturating CPU has on request latency.
```

Not sure about this one but trying to give a feel for why it's different to build intuition rather than just state it.